core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/util/HttpChecksumUtils.java

```diff
@@ -127,4 +140,36 @@ private static boolean isTrailerChecksumPresent(SdkHttpRequest sdkHttpRequest, C
         }
         return false;
     }
+
+    /**
+     * Calculates the content length of a request. If the content-length isn't in the header,
+     * the method reads the whole input stream to get the length.
```

Comment:
probably copy paste, but this method doesn't deal with a request or header, but payload directly.